torch/distributed/_composable/replicate.py

```diff
@@ -25,8 +51,11 @@ def mark_modules(self, *modules: nn.Module, **kwargs) -> None:
         self.kwargs = kwargs
 
     def _recursive_collect_params(self, module: nn.Module) -> None:
-        # TODO: skip if managed by other APIs
+        # skip if managed by other APIs
+        if not _composable(module):
+            return
 
+        # skip if module parameters already collected
         if hasattr(replicate.state(module), "_params_collected"):
             if replicate.state(module)._params_collected:
```

Comment:
nit: Before this PR:
```suggestion
        # skip if module parameters already collected
        if getattr(replicate.state(module), "_params_collected", False):
```

Reply:
Thanks, it's simpler :) But with L62, it seems we can't do this?

Reply:
Oh sorry, I did not see below. Definitely keep it as is :)